netcore/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -322,16 +325,16 @@ public static unsafe Array CreateInstance (Type elementType, int[] lengths, int[
 			if (runtimeType == null)
 				ThrowHelper.ThrowArgumentException (ExceptionResource.Arg_MustBeType, ExceptionArgument.elementType);
 
-			Array array;
+			Array array = null;
 			fixed (int* pLengths = &lengths [0])
 			fixed (int* pLowerBounds = &lowerBounds [0])
-				array = InternalCreate (runtimeType._impl.Value, lengths.Length, pLengths, pLowerBounds);
+				InternalCreate (ref array, runtimeType._impl.Value, lengths.Length, pLengths, pLowerBounds);
 			GC.KeepAlive (runtimeType);
 			return array;
 		}
 
 		[MethodImpl (MethodImplOptions.InternalCall)]
-		static extern unsafe Array InternalCreate (IntPtr elementType, int rank, int* pLengths, int* pLowerBounds);
+		static extern unsafe void InternalCreate (ref Array result, IntPtr elementType, int rank, int* lengths, int* lowerBounds);
```

Comment:
Shouldn't this be `out Array`?

Reply:
No. It should be ref.
Out is a micro optimization and
I frequently see native code fail to set out parameters, violating safety and correctness. Out is to be avoided.